file/writable_file_writer.cc

```diff
@@ -407,7 +408,15 @@ IOStatus WritableFileWriter::WriteBuffered(const char* data, size_t size) {
       {
         auto prev_perf_level = GetPerfLevel();
         IOSTATS_CPU_TIMER_GUARD(cpu_write_nanos, env_);
-        s = writable_file_->Append(Slice(src, allowed), IOOptions(), nullptr);
+        if (perform_data_verification_) {
+          std::string checksum;
```

Comment:
Can we reuse it for the whole file, rather than recreating a string for every write?

Reply:
Sure